src/main.cpp

```diff
@@ -1,40 +1,99 @@
 #include "btserialapp.h"
+#include "btleserialserver.h"
 
 #include <QCoreApplication>
+#include <QCommandLineParser>
+#include <QMap>
+#include <QMetaEnum>
 
 #include <signal.h>
 #include <unistd.h>
 
 void catchUnixSignals(const std::vector<int>& quitSignals,
                       const std::vector<int>& ignoreSignals = std::vector<int>()) {
 
-    auto handler = [](int sig) ->void {
-        printf("\nquit the application (user request signal = %d).\n", sig);
-        QCoreApplication::quit();
-    };
+  auto handler = [](int sig) ->void {
+    printf("\nquit the application (user request signal = %d).\n", sig);
+    QCoreApplication::quit();
+  };
 
-    // all these signals will be ignored.
-    for ( int sig : ignoreSignals )
-        signal(sig, SIG_IGN);
-
-    // each of these signals calls the handler (quits the QCoreApplication).
-    for ( int sig : quitSignals )
-        signal(sig, handler);
+  // all these signals will be ignored.
+  for ( int sig : ignoreSignals ){
+    signal(sig, SIG_IGN);
+  }
+  // each of these signals calls the handler (quits the QCoreApplication).
+  for ( int sig : quitSignals ){
+    signal(sig, handler);
+  }
 }
 
 
 //owner id, device id
 int main(int argc, char *argv[])
 {
-  qDebug("hello!");
   QCoreApplication a(argc, argv);
+  QCoreApplication::setApplicationName("btSerialServer");
+  QCoreApplication::setApplicationVersion("0.1");
+
+  QCommandLineParser parser;
+  parser.setApplicationDescription("btSerialServer");
+  parser.addHelpOption();
+  parser.addVersionOption();
+  
+  QCommandLineOption actorIDOption(QStringList() << "a" << "actor-id",
+        QCoreApplication::translate("main", "Actor ID in Hex (XX:XX...)"),
```

Comment:
do you actually expect `:` between octets?

Reply:
The parser is such that it will ignore any non-hex characters in the string, so colons or no colons is fine.

Reply:
Got it, should probably make documentation most closely match the typical usage.... which is mongodb ID's that are just copy-pasted into place.